openquake/smt/residuals/residual_plotter.py

```diff
@@ -1224,8 +1236,17 @@ def WeightsTable(residuals,filename):
     model_weights_avg_df.columns=model_weights_columns
     final_model_weights_df=pd.concat([model_weights_df,model_weights_avg_df])
     
-    final_model_weights_df.to_csv(filename,sep=',')
-    display(final_model_weights_df)
+    # Table and display outputs (need to assign simplified GMPE names)
+    final_model_weights_df_output = final_model_weights_df
+    model_weights_columns_all_output={}
+    for gmpe in range(0,len(residuals.gmpe_list)):
+         model_weights_columns_all_output[gmpe] = str(residuals.gmpe_list[
```

Comment:
This line is a bit confusing. To increase readability, first I would save the string in a tmp variable and then do the rest.
```
tmp = str(residuals.gmpe_list[gmpe_list_series[gmpe]])
model_weights_columns_all_output[gmpe] = tmp.split('(')[0].replace('\n',', ') + ' LLH'
```

Reply:
Same above at line 1174

Reply:
There are many points in your PR where this suggestion could be also applied.